docs/data/data-grid/pivoting/pivoting.md

```diff
@@ -2,16 +2,138 @@
+
+```tsx
+<DataGridPremium
+  initialState={{ pivoting: { model: pivotModel, active: true, panelOpen: true } }}
+/>
+```
+
+{{"demo": "GridPivotingInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Controlled pivoting
+
+For fully controlled pivoting state, you can use the following props:
+
+- Pivot model:
+  - `pivotModel`: Controls the current pivot configuration.
+  - `onPivotModelChange`: Callback fired when the pivot model changes.
+- Pivot mode toggle:
+  - `pivotActive`: Controls whether pivot mode is active.
+  - `onPivotActiveChange`: Callback fired when the pivot mode changes between active and inactive.
+- Pivot panel:
+  - `pivotPanelOpen`: Controls whether the pivot panel is open.
+  - `onPivotPanelOpenChange`: Callback fired when the pivot panel is opened or closed.
+
+{{"demo": "GridPivotingControlled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Using fields in the pivot model multiple times
+
+While this is not supported yet, we are working to bring this feature to the Data Grid.
+
+## Disable pivoting
+
+To disable pivoting, set the `disablePivoting` prop to `true`:
+
+```tsx
+<DataGridPremium disablePivoting />
+```
+
+## Derived columns in pivot mode
+
+In pivot mode, it's often useful to group data by a year or quarter.
+The Data Grid automatically generates year and quarter columns for each **Date** column for this purpose.
+
+For example, the sales dataset we've been using throughout the examples has a **Quarter** column.
+But often in real-world datasets, each sale record would have a precise **Transaction Date** field, like in the following demo.
+
+The **Transaction Date** column is represented by additional columns in pivot mode: **Transaction Date (Year)** and **Transaction Date (Quarter)**:
+
+{{"demo": "GridPivotingFinancial.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom derived columns
+
+Use the `getPivotDerivedColumns` prop to customize derived columns.
+This prop is called for each original column and returns an array of derived columns, or `undefined` if no derived columns are needed.
+
+{{"demo": "GridGetPivotDerivedColumns.js", "bg": "inline", "defaultCodeOpen": true}}
+
+## Sticky column groups
+
+Depending on the pivot mode, some column groups might exceed the width of the Data Grid viewport.
+To improve the user experience, you can make these column groups "sticky" so that the column group labels remain visible while scrolling horizontally.
+
+You can use the `sx` prop to apply the necessary styles:
+
+```tsx
+<DataGridPremium
+  sx={{
+    '& .MuiDataGrid-columnHeader--filledGroup': {
+      '& .MuiDataGrid-columnHeaderTitleContainer': {
+        overflow: 'visible',
+      },
+      '& .MuiDataGrid-columnHeaderTitleContainerContent': {
+        position: 'sticky',
+        left: 8,
+      },
+    },
+  }}
+/>
+```
+
+{{"demo": "GridPivotingMovies.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Advanced demo
```

Comment:
It would be nice to have a sentence or two here to give the reader more context about what's going on in the demo.